siliconcompiler/core.py

```diff
@@ -4987,6 +5037,130 @@ def __getstate__(self):
         attributes['modules'] = {}
         return attributes
 
+    #######################################
+    def _generate_testcase(self, step, index, archive_name, include_pdks=True, include_libraries=True, hash_files=False):
+        # Save original schema since it will be modified
+        schema_copy = self.schema.copy()
+
+        additional_files = {}
+
+        issue_dir = tempfile.TemporaryDirectory(prefix='sc_issue_')
+
+        self.set('option', 'continue', True)
+        if hash_files:
+            for key in self.allkeys():
+                if key[0] == 'history':
+                    continue
+                if 'file' not in self.get(*key, field='type'):
+                    continue
+                for _, key_step, key_index in self.schema._getvals(*key):
+                    self.hash_files(*key, step=key_step, index=key_index)
+
+        manifest_path = os.path.join(issue_dir.name, 'manifest.json')
+        self.write_manifest(manifest_path)
+        additional_files[manifest_path] = 'manifest.json'
+
+        flow = self.get('option', 'flow')
+        tool, task = self._get_tool_task(step, index, flow=flow)
+
+        # Set copy flags for _collect
```

Comment:
I'm a little worried about the hard-coded groups of files to copy in, seems like it could be brittle (although doing them as major schema categories helps). I think 2 things that may be missing are `['option', 'cmdfile']` and `['fpga', 'arch']`. 

 I wonder if doing this by exclusion would be safer, i.e. set all file types to copy, except a few hardcoded paths that are unnecessary/marked as exclude by the user (I imagine 'package' stuff could be bloat?)

Reply:
reworked, lets see if that is better